srcdeps-maven-plugin/src/main/java/org/srcdeps/mvn/plugin/ExtensionsXmlUtils.java

```diff
@@ -0,0 +1,164 @@
+            this.artifactId = artifactId;
+        }
+
+        public boolean matches(String groupId, String artifactId) {
+            return this.groupId.equals(groupId) && this.artifactId.equals(artifactId);
+        }
+    }
+
+    private static String getFirstChildTextValue(Element parent, String childName) {
+        NodeList children = parent.getChildNodes();
+        for (int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            if (child instanceof Element) {
+                Element elem = (Element) child;
+                if (childName.equals(elem.getNodeName())) {
+                    return getOrCreateFirstTextChild(elem).getTextContent();
+                }
+            }
+        }
+        return null;
+    }
+
+    private static Text getOrCreateFirstTextChild(Element node) {
+        NodeList children = node.getChildNodes();
+        for (int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            if (child instanceof Text) {
+                return (Text) child;
+            }
+        }
+        Text result = node.getOwnerDocument().createTextNode("");
+        node.appendChild(result);
+        return result;
+    }
+
+    private static void setTextValue(Element parent, String childName, String childValue) {
+        Element elem = parent.getOwnerDocument().createElement(childName);
+        parent.appendChild(elem);
+        Text textNode = parent.getOwnerDocument().createTextNode(childValue);
+        elem.appendChild(textNode);
+    }
+
+    /**
+     * Adds or upgrades to the given {@code newVersion} the srcdeps extensions listed in {@link SrcdepsExtension} in the
+     * given DOM {@code document}.
+     *
+     * @param doc
+     *            the DOM of {@code extensions.xml} to modify, may be an empty DOM
+     * @param newVersion
+     *            the srcdeps version to add or upgrade to
+     */
+    public static void upgradeOrAddSrcdeps(Document doc, String newVersion) {
+
+        Element extensions = doc.getDocumentElement();
+        if (extensions == null) {
+            extensions = doc.createElement(ExtensionsXmlElement.extensions.name());
+            doc.appendChild(extensions);
+        }
+
+        for (SrcdepsExtension srcdepsExtension : SrcdepsExtension.values()) {
+
+            boolean extensionFound = false;
+            NodeList extChildren = extensions.getChildNodes();
+            for (int i = 0; i < extChildren.getLength(); i++) {
+                Node extChild = extChildren.item(i);
+                if (extChild instanceof Element) {
+                    Element extension = (Element) extChild;
+                    if (ExtensionsXmlElement.extension.name().equals(extChild.getNodeName())) {
+                        final String groupId = getFirstChildTextValue(extension, ExtensionsXmlAttribute.groupId.name());
+                        final String artifactId = getFirstChildTextValue(extension,
```

Comment:
Just out of curiosity: what's the rule to decide on whether to add `final` or not? I usually trust the compiler to do the right thing for me, but I wonder if I should start explicitly adding `final` to my code